### DISCUSSION TITLE
A way to make nuclei sleep or delay matchers check?

### DISCUSSION THREAD

**Discussion by YotizJ13** [projectdiscovery/nuclei-templates#11181] [Q&A] [answered]
Hi,
I made a template to check for a callback to the interactsh server, but it happends in a delay of approximately 5 seconds. Till the callback is made, nuclei already closes the scan with the 'No results found' message.
Is there a way to delay the matchers check, or at least wait up to 10 seconds in order to check the callback?

Thanks.

**Comment by ehsandeep** [comment, marked as answer]
@YotizJ13 

```
   -interactions-cooldown-period int    extra time for interaction polling before exiting (default 5)
```

**Reply by YotizJ13** [reply]
Wow that was fast. Can't believe I missed that one, thanks!

**Reply by YotizJ13** [reply]
Another question - Can I define it in the template?

**Reply by ehsandeep** [reply]
You can use `wait_for()` helper in request, see https://cloud.projectdiscovery.io/?template=CVE-2023-47246 as example template where it used `{{wait_for(9)}}` to wait for 9 sec before making 2nd request, also ensure to increase `@timeout` to avoid context deadline error as default timeout is 10 sec.

**Reply by YotizJ13** [reply]
Thank you so much! Closing the discussion.
